unison-cli/src/Unison/Codebase/Editor/HandleCommand.hs

```diff
@@ -282,6 +282,20 @@ commandLine config awaitInput setBranchRef rt sdbxRt notifyUser notifyNumbered l
                 Codebase.putWatch codebase kind hash value'
           pure $ Right rs
 
+typecheckCommand ::
+  (Random.DRG gen, Monad m) =>
+  Codebase m Symbol Ann ->
+  [Type Symbol Ann] ->
+  NamesWithHistory.NamesWithHistory ->
+  SourceName ->
+  LexedSource ->
+  gen ->
+  m (TypecheckingResult Symbol)
+typecheckCommand codebase ambient names sourceName source gen = do
```

Comment:
Pulling this out into a separate command allows us to call the same typechecker path in both UCM and the LSP.